computations.py

```diff
@@ -0,0 +1,140 @@
+        minimum_temperature_record_in_year = []
+        try:
+            for weather_monthly_records in weather_records[year].keys():
+                weather_daily_records = weather_records[year][weather_monthly_records]
+                minimum_temperature_record_in_month = min(
+                    weather_daily_records, key=lambda x: x[minimum_temperature] if x[minimum_temperature] is not None
+                    else float('inf'))
+                if minimum_temperature_record_in_month[minimum_temperature] < minimum_temperature_in_year:
+                    minimum_temperature_in_year = minimum_temperature_record_in_month[minimum_temperature]
+                    minimum_temperature_record_in_year = minimum_temperature_record_in_month
+            minimum_temperature_month, minimum_temperature_day = Computations.get_month_and_day_from_date(
+                minimum_temperature_record_in_year[date])
+            return minimum_temperature_month, minimum_temperature_day, minimum_temperature_in_year
+        except KeyError:
+            sys.exit('Weather Record not exist for year: {}'.format(year))
+
+    @staticmethod
+    def find_maximum_humidity_in_year(year, all_weather_records):
+        weather_attributes, weather_records = all_weather_records
+        maximum_humidity = Computations.get_weather_attribute_index(weather_attributes, "Max Humidity")
+        date = Computations.get_weather_attribute_index(weather_attributes, "PKT")
+        max_humidity_in_year = float("-inf")
+        max_humidity_record_in_year = []
+        try:
+            for weather_monthly_records in weather_records[year].keys():
+                weather_daily_records = weather_records[year][weather_monthly_records]
+                max_humidity_record_in_month = max(
+                    weather_daily_records, key=lambda x: x[maximum_humidity] if x[maximum_humidity] is not None
+                    else float('-inf'))
+                if max_humidity_record_in_month[maximum_humidity] > max_humidity_in_year:
+                    max_humidity_in_year = max_humidity_record_in_month[maximum_humidity]
+                    max_humidity_record_in_year = max_humidity_record_in_month
+            max_humidity_month, max_humidity_day = Computations.get_month_and_day_from_date(
+                max_humidity_record_in_year[date])
+            return max_humidity_month, max_humidity_day, max_humidity_in_year
+        except KeyError:
+            sys.exit('Weather Record not exist for year: {}'.format(year))
+
+    @staticmethod
+    def find_average_highest_temperature_in_month(year, month, all_weather_records):
+        weather_attributes, weather_records = all_weather_records
+        maximum_temperature = Computations.get_weather_attribute_index(weather_attributes, "Max TemperatureC")
+        average_highest_temperature_in_month = 0
+        try:
+            no_of_weather_records_in_month = len(weather_records[year][month])
```

Comment:
We are using this variable at single instance in this function, what is the point of declaring it? Let directly call len at that point.